### PR TITLE
automatically replacing designated function calls 

### PR DESCRIPTION
**Pull request by theseanl**

### CODE REVIEW

File: deobfuscator/comm.js [hunk 1]
```diff
@@ -201,29 +233,49 @@ var SandboxEval = function () {
...
-        else if(evt.sucess == 0) {
-            reject(evt.data.error);
-        }
-        else {
-            reject(Error("Internal error."));
-        }
-    }
+    this.eval = function(code) {
+        return sandboxLoaded.catch(function(error){
+            ;
```
Comment (ameshkov) [inline deobfuscator/comm.js]: At least log it

File: deobfuscator/comm.js [hunk 2]
```diff
@@ -201,29 +233,49 @@ var SandboxEval = function () {
...
+        }).then(function() {
+            return new Promise(function(resolve, reject) {
+                var thisId = (new Date()).getTime() * 10 + Math.floor(Math.random() * 10); // Or should I store request Ids in a closure and increase it like 1,2,3,4..?
```
Comment (ameshkov) [inline deobfuscator/comm.js]: It would be much better to have an auto-incremental ID

**Review by ameshkov — COMMENTED** [review]
Code is a bit messy, but that's quite okay for its purpose

File: deobfuscator/comm.js (hunk 1 above)
Comment (theseanl) [reply deobfuscator/comm.js]: I also think that it is messy, I'd appreciate if you could provide some advice on how I can improve it.

**Pull request merged**